### PR TITLE
Feat/lesson07

### LINKED ISSUE #28: Allow get_dataset to create the data directory if it does not exist

### PR DESCRIPTION
**Pull request by lewtun**
* Badges will be added later
* Solution will be removed later

Fixes #28

### CODE REVIEW

File: notebooks/00_core.ipynb [hunk 1]
```diff
@@ -31,7 +31,11 @@
     "import numpy as np\n",
```
Comment (lvwerra) [inline notebooks/00_core.ipynb:31]: <p>care to add a docstring?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='46'/>

Comment (lvwerra) [inline notebooks/00_core.ipynb:31]: <p>care to add a docstring?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='48'/>

File: notebooks/01_structured.ipynb [hunk 2]
```diff
@@ -71,7 +71,8 @@
     "from sklearn.tree import export_graphviz\n",
```
Comment (lvwerra) [inline notebooks/01_structured.ipynb:71]: <p>do you need that import?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

File: notebooks/lesson07_classification.ipynb [hunk 3]
```diff
@@ -0,0 +1,3932 @@
+{
```
Comment (lvwerra) [inline notebooks/lesson07_classification.ipynb:1]: <p>I don't know if you plan to do this in the slides, but I think it would be good to underline the difference between regression and classification again. maybe also state again what the leaf in a regression and classification forest yield. just because last year we had some confusion about this.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

Comment (lvwerra) [inline notebooks/lesson07_classification.ipynb:1]: <p>This assumes that you already know that " " is in the dataset. Maybe it would be a good opportunity to show the students how they could investigate/debug the dataset (maybe with a value counter?).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (lvwerra) [inline notebooks/lesson07_classification.ipynb:1]: <p>is there a comment missing before grid search?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='92'/>

Comment (lvwerra) [inline notebooks/lesson07_classification.ipynb:1]: <p>this plot does not seem very informative. maybe delete it?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='105'/>

Comment (lvwerra) [inline notebooks/lesson07_classification.ipynb:1]: <p>can you explain why the encoding has an influence on the performance?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='119'/>

Comment (lvwerra) [inline notebooks/lesson07_classification.ipynb:1]: <p>This had no influence on the performance, right? You could mention that you didn't categorise tenure for performance improvement but for interpretability.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='148'/>

Comment (lvwerra) [inline notebooks/lesson07_classification.ipynb:1]: <p>since it is a central concept, you could maybe repeat what these numbers mean? maybe rephrase it in terms of "if our model classifies a customer as churning it is x% of the time correct. if it says the customer is not churning it is y% of the time wrong." maybe mention that these are very central questions for example in drug development and medicine. maybe you could also show the unnormalized confusion matrix (otherwise it is hard to compare elements in columns).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='152'/>

**Comment by lvwerra** [comment]
looks good overall!

File: notebooks/00_core.ipynb (hunk 1 above)
Comment (lewtun) [reply notebooks/00_core.ipynb:31]: <p>done!</p>

File: notebooks/01_structured.ipynb (hunk 2 above)
Comment (lewtun) [reply notebooks/01_structured.ipynb:71]: <p>which one? <code>pandas</code>? </p>

File: notebooks/lesson07_classification.ipynb (hunk 3 above)
Comment (lewtun) [reply notebooks/lesson07_classification.ipynb:1]: <p>yeah i'll do this on the ipad during the lesson</p>

Comment (lewtun) [reply notebooks/lesson07_classification.ipynb:1]: <p>good idea - done!</p>

Comment (lewtun) [reply notebooks/lesson07_classification.ipynb:1]: <p>good catch - fixed!</p>

Comment (lewtun) [reply notebooks/lesson07_classification.ipynb:1]: <p>done!</p>

Comment (lewtun) [reply notebooks/lesson07_classification.ipynb:1]: <p>done!</p>

Comment (lewtun) [reply notebooks/lesson07_classification.ipynb:1]: <p>thanks!</p>

Comment (lewtun) [reply notebooks/lesson07_classification.ipynb:1]: <p>thanks - included!</p>
